src/backend/storeManagers/gog/library.ts

```diff
@@ -443,6 +445,10 @@ export async function refresh(): Promise<ExecResult> {
     }
   }
 
+  for (const game of gamesObjects) {
```

Comment:
maybe, instead of checking `addSteamShortcutsUninstalled` for every game in `addShortcuts`, you could check that once here and then if it's try you do the loop

Reply:
also, won't this trigger one system notification per game? it can be really annoying to turn this on, restart heroic and start seeing maybe hundreds of notifications popping up

Reply:
there's also A LOT of work happening for every game: getting wiki game info, downloading the images, checking if steam files exist, etc

I think using `addShortcuts` in a loop is not scalable (there's users with 1000+ games), we would need a different method, something that adds all games and then saves the shortcuts.vdf once

Reply:
also, having each library running its own `for` loop and all of them writing to the `shortcuts.vdf` file at the same time in parallel can create a bunch of race conditions (what if while the gog loops is trying to add a game, the legendary loop saved a file, then the gog loops overrides it?)

or if I refresh the library while the games are being added, then this would run again for the same store in parallel

Reply:
finally, I don't think it's scalable to wait for all the shortcuts to be added to continue with the refresh call, this will delay showing the library in heroic